src/app/_services/api.service.ts

```diff
@@ -10,58 +10,71 @@ const baseUrl = environment.apiUrl;
 @Injectable()
 export class ApiService {
 
-  constructor(private http: HttpClient) { }
+    constructor(private http: HttpClient) { }
 
-  getHttpHeaders() {
-  	let rtn = {
-  		headers: new HttpHeaders({
-  			'Content-Type': 'application/json'
-  		})
-  	}
+    getHttpHeaders() {
+          let rtn = {
+                  headers: new HttpHeaders({
+                          'Content-Type': 'application/json'
+                  })
+          }
 
-  	return rtn;
-  }
+          return rtn;
+    }
 
-  // http.get needs an Authorization in header
-  getHttpHeadersWithBase64Authorization(username:string, password: string) {
-  	let rtn = {
-  		headers: new HttpHeaders({
-        'Authorization': "Basic " + btoa(`${username}:${password}`),
-  			'Content-Type': 'application/json'
-  		})//,
-      //responseType: 'json'
-  	}
+    // http.get needs an Authorization in header
+    getHttpHeadersWithBase64Authorization(username:string, password: string) {
+          let rtn = {
+                  headers: new HttpHeaders({
+          'Authorization': "Basic " + btoa(`${username}:${password}`),
+                          'Content-Type': 'application/json'
+                  })//,
+        //responseType: 'json'
+          }
 
-  	return rtn;
-  }
+          return rtn;
+    }
 
-  get(url:string, username:string, password:string) {
-    return this.http.get<any>(baseUrl + url,
-      this.getHttpHeadersWithBase64Authorization(username, password))
-      .pipe(
-        catchError(this.handleError)
-      );
-  }
+    get(url:string, username:string, password:string) {
+      return this.http.get<any>(baseUrl + url,
+        this.getHttpHeadersWithBase64Authorization(username, password))
+        .pipe(
+          catchError(this.handleError)
+        );
+    }
 
-  post(url, data) {
-  	return this.http.post(baseUrl + url, data, this.getHttpHeaders())
-  }
+/* 
+    post(url, data) {
+          return this.http.post(baseUrl + url, data, this.getHttpHeaders())
+    }
+*/
+      
+    post(url:any, data:any) {
 
-  private handleError(error: HttpErrorResponse) {
-    if (error.error instanceof ErrorEvent) {
-      // A client-side or network error occurred. Handle it accordingly.
-      console.error('An error occurred:', error.error.message);
-    } else {
-      // The backend returned an unsuccessful response code.
-      // The response body may contain clues as to what went wrong,
-      console.log("In handleError: error= " + JSON.stringify(error));
-      console.error(
-        `Backend returned code ${error.status}, ` +
-        `body was: ${error.error}`);
+      var username = "johnathan";
+      var password = "password";
+      // alert(baseUrl + "" + url);
+      // return this.http.post(baseUrl + url, data, this.getHttpHeadersWithBase64Authorization(username, password))
+      return this.http.post('http://localhost:8080/api/question', data, this.getHttpHeadersWithBase64Authorization(username, password));
```

Comment:
Read this line carefully. Remember this is going to Production, live on the internet. What is wrong with this line?